logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/RemoveMemberFromConversationUseCase.kt

```diff
@@ -0,0 +1,29 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.conversation.MLSConversationRepository
+import com.wire.kalium.logic.data.conversation.ProtocolInfo
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.flatMap
+
+interface RemoveMemberFromConversationUseCase {
+    suspend operator fun invoke(conversationId: ConversationId, userIdList: List<UserId>)
+}
+
+class RemoveMemberFromConversationUseCaseImpl(
+    private val conversationRepository: ConversationRepository,
+    private val mlsConversationRepository: MLSConversationRepository,
+) : RemoveMemberFromConversationUseCase {
+    override suspend fun invoke(conversationId: ConversationId, userIdList: List<UserId>) {
+        conversationRepository.detailsById(conversationId).flatMap { conversation ->
+            when (conversation.protocol) {
+                is ProtocolInfo.Proteus ->
+                    conversationRepository.deleteMembers(userIdList, conversationId)
+
+                is ProtocolInfo.MLS ->
+                    mlsConversationRepository.removeMembersFromMLSGroup(conversation.protocol.groupId, userIdList)
```

Comment:
`MlsConversationRepository` is currently structured as an internal dependency to the `ConversationRepository` so it would make more sense to have this `when` statement inside a method on the  `ConversationRepository`.